### ISSUE TITLE
Result Extraction

### ISSUE THREAD

**Issue by JiaLim98** [zzzxxxttt/pytorch_simple_centernet_45#28]
Hi @zzzxxxttt,

Thank you for your amazing work. I have completed my training and testing. However, I require two more results for documentation purposes.
1. Precision-Recall Curve: May I know which python file is the repo using? This is because I found two cocoeval.py after finish 
    compiling PythonAPI, under the two paths stated below. I have tried modifying both but can't seem to work (looks like the 
    test.py is not using both) . Can you let me know how can I extract the 101 datapoints for PR curve?
    - \lib\cocoapi\PythonAPI\build\lib.linux-x86_64-3.6\pycocotools, and
    - \lib\cocoapi\PythonAPI\pycocotools
2. mAP@0.5 by classes. I can't seem to find the line of code that gives me this. Can you let me know how? I got the overall coco results already, I just need the results by classes.

Hope to hear from you soon. Thank you.

Cheers,
JiaLim98

**Comment by zzzxxxttt** [comment]
For the pr curve, [this issue](https://github.com/facebookresearch/maskrcnn-benchmark/issues/353) may help, and detectron2 generates class-wise mAP during evaluation, you can check the implementation of it.

**Comment by JiaLim98** [comment]
Thanks for the reply.

For anyone who wants to output class-wise mAP, this is the most straightforward way I found.
The mAP for entire model is calculated from the average of APs for each classes. I have found the code for exporting the class-wise mAP from cocoeval.py in Pycocotools [here](https://github.com/kimyoon-young/centerNet-deep-sort/blob/master/tools/cocoeval.py) from line 458 to line 464.

**Comment by yangyang117** [comment]
Jialim，I am very happy that you have completed the training and testing work, may I know your final accuracy? I am trying to train this net in my datasets, but I still not complete it.

**Comment by JiaLim98** [comment]
Hi @yangyang117,

My final accuracy (without finetuning) are:
mAP@0.5 = 92.02% and mAP@0.5:0.05:0.95 = 40.60%
Hope it helps.

**Comment by yangyang117** [comment]
Thanks  for your answering,  is this for your own datasets or public datasets?

**Comment by JiaLim98** [comment]
It is an open-sourced PCB defects dataset.

**Comment by yangyang117** [comment]
Thank you, it's very helpful.

**Comment by 18342960922** [comment]
> 它是开源的PCB缺陷数据集。

您好，麻烦问一下 自己的数据集 遇到这个问题 No such file or directory: './data/voc/VOCdevkit/VOC2007/annotations_cache'该如何处理

**Comment by JiaLim98** [comment]
您可以检查文件目录是否确实存在. 对于第一次训练，这个问题不应该存在.
